File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/J.Факторизация/solution.cpp
#include <iostream>
#include <vector>


void solution(long long num) {
    /*
    Time Complexity O(sqrt(N)log(sqrt(N)))
    Memory Complexity O(sqrt(N))
    */
    std::vector<long long> output;
    while (num % 2 == 0) {
        output.push_back(2);
        num /= 2;
    }
    for (long long mod = 3; mod * mod <= num; mod += 2) {
        while (num % mod == 0) {
            output.push_back(mod);
            num /= mod;
        }
    }
    if (num > 2) {
        output.push_back(num);
    }
    int n = output.size();
    for (int index = 0; index < n; ++index) {
        if (index) {
            std::cout << " ";
        }
        std::cout << output[index];
    }
    std::cout << "\n";
}


int main() {
    long long num;
    std::cin >> num;
    solution(num);
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/J.Факторизация/solution.py
import sys


def solution(n: int) -> None:
    """
    Time Complexity O(sqrt(N)log(sqrt(N)))
    Memory Complexity O(sqrt(N))
    """
    factorize: list[str] = []
    while n % 2 == 0:
        factorize.append('2')
        n //= 2
    upper_bound: int = int(n**.5) + 1
    for prime in range(3, upper_bound, 2):
        while n % prime == 0:
            factorize.append(str(prime))
            n //= prime
    if n > 2:
        factorize.append(str(n))
    factorize_n: int = len(factorize)
    for idx in range(factorize_n):
        if idx:
            sys.stdout.write(' ')
        sys.stdout.write(str(factorize[idx]))
    sys.stdout.write('\n')


if __name__ == '__main__':
    n: int = int(sys.stdin.readline().rstrip())
    solution(n)